src/Video.tsx

```diff
@@ -606,14 +613,102 @@ const Video = forwardRef<VideoRef, ReactVideoProps>(
         : ViewType.SURFACE;
     }, [drm, useSecureView, useTextureView, viewType]);
 
+    const _renderPoster = useCallback(() => {
+      // poster resize mode
+      let _posterResizeMode: ImageResizeMode = 'contain';
+
+      if (!isPosterDeprecated && poster?.resizeMode) {
+        _posterResizeMode = poster.resizeMode;
+      } else if (posterResizeMode && posterResizeMode !== 'none') {
+        _posterResizeMode = posterResizeMode;
+      }
+
+      // poster style
+      const baseStyle: StyleProp<ImageStyle> = {
+        ...StyleSheet.absoluteFillObject,
+        resizeMode: _posterResizeMode,
+      };
+
+      let posterStyle: StyleProp<ImageStyle> = baseStyle;
+
+      if (!isPosterDeprecated && poster?.style) {
+        const styles = Array.isArray(poster.style)
+          ? poster.style
+          : [poster.style];
+        posterStyle = [baseStyle, ...styles];
+      }
+
+      // render poster
+      if (renderLoader && poster) {
+        console.warn(
+          'You provided both `renderLoader` and `poster` props. Poster is ignored.',
+        );
+      }
+
+      if (renderLoader) {
+        const Loader = renderLoader as unknown as MemoExoticComponent<FC>;
+
+        const compType =
+          Loader?.$$typeof?.toString() === 'Symbol(react.memo)'
+            ? 'memoComp'
+            : 'comp';
+
+        const isFunctionComp = typeof renderLoader === 'function';
+
+        // check if valid jsx
+        if (
+          React.isValidElement(
+            {
+              memoComp: Loader,
+              comp: isFunctionComp ? renderLoader(null) : renderLoader,
+            }[compType],
+          )
+        ) {
+          console.warn(
+            'Invalid renderLoader component. Please provide a valid JSX component',
+          );
+          return;
+        }
+
+        return {
+          memoComp: (
+            <View style={StyleSheet.absoluteFill}>
+              <Loader />
+            </View>
+          ),
```

Comment:
Enable to use memoized component (React.memo(Comp)) as renderLoader={MemoComp} (earlier possible only renderLoader={<MemoComp/>})

Reply:
This looks quite impressive but let's just change type of renderLoader to `React.ReactNode` and it should be fine I guess without this code 🥲